Dynamix/MainActivity.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using Android.App;
+using Android.OS;
+using Android.Runtime;
+using Android.Views;
+using Android.Widget;
+
+namespace Dynamix
+{
+	[Activity (Label = "Dynamix", MainLauncher = true)]
+	public class MainActivity : Activity
+	{
+		private TextView textField;
+		private Button theButton;
+
+		protected override void OnCreate (Bundle savedInstanceState)
+		{
+			base.OnCreate (savedInstanceState);
+
+			SetContentView (Resource.Layout.activity_main);
+
+			textField = FindViewById<TextView> (Resource.Id.textField);
+			theButton = FindViewById<Button> (Resource.Id.theButton);
+
+			theButton.Click += OnButtonClick;
+		}
+
+		private void OnButtonClick (object sender, EventArgs eventArgs)
+		{
+			dynamic calculatorConstructor = new Java.Interop.Dynamic.DynamicJavaClass ("com/xamarin/dynamix/Calculator");
+
+			var calculator = calculatorConstructor ();
+
+			// normal
+			var answer = calculator.add (3, 6);
+			textField.Text += $"\nNORMAL: 3 + 6 = {answer}";
+
+			// use a callback
+			var listener = new MyListener (textField);
+			calculator.add (2, 4, (IAnswerListener) listener);
+		}
+
+		// this is to create the interface on the Java side
+		[Register ("com/xamarin/dynamix/AnswerListener")]
+		public interface IAnswerListener : IJavaObject { }
+
+		public class MyListener : Java.Lang.Object, IAnswerListener
```

Comment:
Create a new type, and inherit from `Java.Lang.Object` so we jet the Java code generated. Make sure to inherit that new interface that was just registered so this class inherits that interface on the Java side as well.

Reply:
Alternate take:

@jpobst: if we had a `JniImplementsAttribute`, we could instead do:

```csharp
[JniImplements("com/xamarin/dynamix/AnswerListener")]
class MyListener : Java.Lang.Object {
    [Export]
    public void onCalculated (int value) {/* … */ }
}
```

This would also avoid the need to "bind" `IAnswerListener` in a fashion which isn't a *real* binding.

Reply:
A benefit to `JcwImplementsAttribute` is that it suggests we should also have a `JcwExtendsAttribute`, which would allow a managed type to explicitly control it's *base* class, without requiring that the base class be bound.

This would be useful if e.g. `AnswerListener` weren't an interface but were instead a class:

```csharp
[JcwExtends("some.randome.BaseClass")]
[JcwImplements("some.randome.Interface")]
public class MyExample : Java.Lang.Object {
    // ...
}
```

This raises an alternate question: assuming we introduce a set of `Jcw*` attributes, in what namespace should they live?  I'm not sure that the `Java.Interop` namespace is still appropriate, but perhaps it is?

Reply:
Considering that `Java.Interop.ExportAttribute` is conceptually the same thing -- add this method to the Java Callable Wrapper class body -- these attributes should likewise be in `Java.Interop`.